newrelic/resource_newrelic_alert_muting_rule_test.go

```diff
@@ -44,6 +96,66 @@ func TestAccNewRelicAlertMutingRule_Basic(t *testing.T) {
 	})
 }
 
+func TestAccNewRelicAlertMutingRule_WithSchedule(t *testing.T) {
+	resourceName := "newrelic_alert_muting_rule.bar"
+	rName := acctest.RandString(5)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckNewRelicAlertMutingRuleDestroy,
+		Steps: []resource.TestStep{
+			// Test: Create
+			{
+				Config: testAccNewRelicAlertMutingRuleWithSchedule(
+					rName,
+					"new muting rule",
+					"product",
+					"EQUALS",
+					"APM",
+					`
+						start_time         = "2021-01-21T15:30:00"
+						end_time           = "2021-01-21T16:30:00"
+						time_zone          = "America/Los_Angeles"
+						repeat             = "WEEKLY"
+						end_repeat         = "2022-06-11T12:00:00"
+						weekly_repeat_days = ["MONDAY", "TUESDAY", "FRIDAY"]
+					`,
+				),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNewRelicAlertMutingRuleExists(resourceName),
+				),
+			},
+			//Test: Update
+			{
+				Config: testAccNewRelicAlertMutingRuleWithSchedule(rName,
+					"updated muting rule schedule",
+					"conditionType",
+					"NOT_EQUALS",
+					"baseline",
+					`
+						start_time         = "2021-02-21T15:30:00"
+						end_time           = "2021-02-21T16:30:00"
+						end_repeat         = "2022-06-11T12:00:00"
+						repeat             = "DAILY"
+						time_zone          = "America/Los_Angeles"
+					`,
+				),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNewRelicAlertMutingRuleExists(resourceName),
+				),
+			},
+			//Test: Import
+			// TODO - determine why this is failing
+			//	{
+			//		ResourceName:      resourceName,
+			//		ImportState:       true,
+			//		ImportStateVerify: true,
+			//	},
+		},
+	})
+}
+
```

Comment:
I think it might be worth adding a test to create a muting rule with a schedule and then remove null out the schedule entirely on update.

Reply:
Yeah, I was just thinking this too! I want to create a bunch of wacky state changes across the acceptance tests like we've been doing by hand locally.

Reply:
Done while pairing.